iothub/device/src/DeviceClient.cs

```diff
@@ -447,9 +449,28 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
         /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
         /// <exception cref="OperationCanceledException">Thrown when the operation has been canceled.</exception>
         /// <returns>The task to await</returns>
+        [Obsolete("This API has been split into three APIs: GetFileUploadSasUri, uploading to blob directly using the Azure Storage SDK, and CompleteFileUploadAsync")]
         public Task UploadToBlobAsync(string blobName, Stream source, CancellationToken cancellationToken) =>
             InternalClient.UploadToBlobAsync(blobName, source, cancellationToken);
 
+        /// <summary>
+        /// Get a file upload SAS URI which the Azure Storage SDK can use to upload a file to blob for this device. See <see href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-file-upload#initialize-a-file-upload">this documentation for more details</see>
+        /// </summary>
+        /// <param name="request">The request details for getting the SAS URI, including the destination blob name.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The file upload details to be used with the Azure Storage SDK in order to upload a file from this device.</returns>
```

Comment:
Returns file upload details or SAS Uri?

Reply:
Should we also mention how long this will be valid?

Reply:
The SAS URI is something that is constructed from the return type here. See the e2e test for details. It looks like this:

Reply:
```
            string blobUriString = String.Format(
                CultureInfo.InvariantCulture,
                "https://{0}/{1}/{2}{3}",
                fileUploadSasUriResponse.HostName,
                fileUploadSasUriResponse.ContainerName,
                Uri.EscapeDataString(fileUploadSasUriResponse.BlobName), // Pass URL encoded device name and blob name to support special characters
                fileUploadSasUriResponse.SasToken);
```

Reply:
This is ideally where I would link the sample that shows this off, but I can't write the sample until this code is checked in since the samples live in a different repo that references this repo

Reply:
Actually, I may as well just add a helper function to do this formatting. Hang tight and I'll push changes for that